grails-app/services/org/pih/warehouse/inboundSortation/InboundSortationService.groovy

```diff
@@ -0,0 +1,56 @@
+package org.pih.warehouse.inboundSortation
+
+import grails.gorm.transactions.Transactional
+import org.pih.warehouse.api.Putaway
+import org.pih.warehouse.api.PutawayItem
+import org.pih.warehouse.api.PutawayStatus
+import org.pih.warehouse.core.ActivityCode
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.receiving.Receipt
+import org.pih.warehouse.receiving.ReceiptItem
+import org.pih.warehouse.shipping.Shipment
+
+@Transactional
+class InboundSortationService {
+
+    def orderIdentifierService
+    def putawayService
+
+    void execute(Receipt receipt) {
+        Shipment shipment = receipt.shipment
+
+        receipt.receiptItems.each { ReceiptItem receiptItem ->
+            if (!receiptItem.binLocation?.supports(ActivityCode.INBOUND_SORTATION)) {
+                log.info"Receipt item destination does not support ${ActivityCode.INBOUND_SORTATION} activity code"
+                return
+            }
+
+            Putaway putaway = createPutaway(shipment)
+            putaway.putawayItems.add(createPutawayItem(receiptItem, shipment))
+            putawayService.savePutaway(putaway)
+        }
+    }
+
+    private Putaway createPutaway(Shipment shipment) {
+        new Putaway(
+                origin: shipment.destination,
+                destination: shipment.destination,
+                putawayNumber: orderIdentifierService.generate(new Order()),
+                putawayAssignee: shipment.createdBy,
+                putawayStatus: PutawayStatus.PENDING
+        )
+    }
+
+    private PutawayItem createPutawayItem(ReceiptItem receiptItem, Shipment shipment) {
+        new PutawayItem(
+                product: receiptItem.product,
+                inventoryItem: receiptItem.inventoryItem,
+                quantity: receiptItem.quantityReceived,
+                recipient: receiptItem.recipient,
+                currentFacility: shipment.destination,
+                currentLocation: receiptItem.binLocation,
+                putawayLocation: receiptItem.product.getInventoryLevel(shipment.destination.id)?.preferredBinLocation,
```

Comment:
This is a good start, but this is going to be much more complicated and probably require its own service. 
For example, let's say we received 100 EA into the Inbound Sortation location and we have outbound demand of
* 10 EA bound for service staging
* 50 EA bound for local delivery staging

We want to have a mechanism that takes the received quantity, calculates outbound demand, determines the preferred putaway location and then generates the appropriate putaway orders. 

So using the strategy pattern, we'd build a putaway context with the relevant input (i.e. 100ea, product A, facility B), pass it through the strategy service and it would create the following putaway orders
* 10 ea to service 
* 50 ea to local delivery 
* 40 ea to preferred bin location

Reply:
I'll start working on that next week.

Reply:
Note that this line might set the putaway location to NULL if there's no preferred bin. I believe a putaway order can be saved without a putaway destination, so as long as we can set it from the putaway UI, then perhaps it's fine to set the putaway destination to NULL here. If not we should find a random bin location (potentially unused, but not necessary) to assign as the destination OR throw an error if one is not found.

Reply:
I've started refactoring code to more generic, with SlottingService, PutawayContext, PutawayStrategy etc.

I will push first draft version today. Then we can extend it next steps.

Regarding putawawy location == null - yes, putaway order can be saved with empty putaway destination with no problems.